extra_data/locality.py

```diff
@@ -0,0 +1,146 @@
+
+def print_counts(fpart):
+    """ Prints the counters of different localities """
+    n_ondisk = len(fpart['NOT_ON_DCACHE']) + len(fpart['ONLINE_AND_NEARLINE']) + len(fpart['ONLINE'])
+    n_ontape = len(fpart['NEARLINE'])
+    n_unavail = len(fpart['UNAVAILABLE'])
+    print(f"{n_ondisk} on disk, {n_ontape} only on tape, {n_unavail} unavailable    ", end='\r')
+    
+def silent(fpart):
+    """ Prints nothing """
+    pass
+
+def partition(files, cb_disp=silent):
+    """ Partition files by locality """
+    fpart = defaultdict(list)
+    for path, loc in list_locality(files):
+        fpart[loc].append(path)
+        cb_disp(fpart)
+    return fpart
+
+def lc_match(files, accept=ONDISK):
+    """ Returns files which has accepted locality """
+    filtered = []
+    for path, loc in list_locality(files):
+        code = DC_LOC_RESP.get(loc, 0)
+        if code & accept:
+            filtered.append(path)
+        else:
+            print(f"Skipping file {path}", file=sys.stderr)
+            print(f"  ({LOCMSG[loc]})", file=sys.stderr)
+            
+    return filtered
+        
+    
+def lc_any(files):
+    """ Returns all files, does nothing """
+    return files
+
+def lc_ondisk(files):
+    """Returns files on disk, excluding any which would be read from tape"""
+    return lc_match(files, ONDISK)
+
+def lc_avail(files):
+    """Returns files which are available on disk or tape
+
+    Excludes files which dCache reports are unavailable.
+    """
+    return lc_match(files, ONTAPE | ONDISK)
+
+def check_dir(basedir):
+    """ Check basedir and prints results """
+    if os.path.isdir(basedir):
+        ls = ( os.path.join(basedir, f) for f in os.listdir(basedir) )
+        files = [ f for f in ls if os.path.isfile(f) ]
+    elif os.path.isfile(basedir):
+        files = [ basedir ]
+    else:
+        files = []
+    
+    print(f"Checking {len(files)} files in {basedir}")
+    fp = partition(files, print_counts)
+    print("")
+    
+    retcode = 0
+    if fp['NEARLINE']:
+        retcode |= 1
+        print("Only on tape:")
+        for file in sorted(fp['NEARLINE']):
+            print(f"  {file}")
+    
+    if fp['UNAVAILABLE']:
+        retcode |= 2
+        print("Unavailable:")
+        for file in sorted(fp['UNAVAILABLE']):
+            print(f"  {file}")
+    
+    unknown_locality = set(fp) - set(DC_LOC_RESP)
+    if unknown_locality:
+        retcode |= 4
+        print("Unknown locality:", unknown_locality)
+        
+    return retcode
+    
+from argparse import ArgumentParser
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv[1:]
+
+    ap = ArgumentParser(prog='extra-data-locality', description="Checks locality of files in the directory")
+    ap.add_argument('path', help="run directory of HDF5 files.")
+    args = ap.parse_args(argv)
+    
+    if not os.path.exists(args.path):
+        print(f"Path '{args.path}' is not found")
+        return -1
```

Comment:
```suggestion
        return 255
```

At least on my system, this appears to get translated to 255. Is it better to do that explicitly? Negative exit codes in some contexts mean a process was terminated by a signal, so it's a bit odd (for me, at least) to see something trying to return a negative exit code.